### PR TITLE
Support for Cosmos - add staking

### PR DESCRIPTION
**Pull request by johnnynanjiang**
## Description

- Add support for Cosmos staking
- Update tests against gaia-13003
- Refactor existing Cosmos support code

## Testing instructions

Tests cases are added in CPP and Swift, test data is from gaia-13003 testnet.

## Types of changes

A new feature for Cosmos staking

### CODE REVIEW

File: swift/Tests/CosmosSignerTests.swift [hunk 1]
```diff
@@ -17,29 +17,60 @@ class CosmosSignerTests: XCTestCase {
         var sendCoinsMessage = TW_Cosmos_Proto_SendCoinsMessage()
         sendCoinsMessage.fromAddress = fromAddress
         sendCoinsMessage.toAddress = "cosmos1zt50azupanqlfam5afhv3hexwyutnukeh4c573"
-        var coin = TW_Cosmos_Proto_Coin()
-        coin.amount = 1
-        coin.denom = "muon"
-        sendCoinsMessage.amount = [coin]
+        var txAmount = TW_Cosmos_Proto_Amount()
```
Comment (hewigovens) [inline swift/Tests/CosmosSignerTests.swift]: checkout `Cosmos+Proto.swift`, we generate swift typealias automatically, so you can use `CosmosAmount`

File: swift/Tests/CosmosSignerTests.swift [hunk 2]
```diff
@@ -17,29 +17,60 @@ class CosmosSignerTests: XCTestCase {
...
+        var feeAmount = TW_Cosmos_Proto_Amount()
+        feeAmount.amount = 200
+        feeAmount.denom = "muon"
 
         var fee = TW_Cosmos_Proto_Fee()
         fee.gas = 200000
-        fee.amount = [feeCoin]
+        fee.amounts = [feeAmount]
 
         var signingInput = TW_Cosmos_Proto_SigningInput()
```
Comment (hewigovens) [inline swift/Tests/CosmosSignerTests.swift]: same here, and we prefer:
```swift
CosmosSigningInput.with {
    $0.chainID = "gaia-13003"
    ....
}
```

Comment (hewigovens) [reply swift/Tests/CosmosSignerTests.swift]: you can apply this pattern for other protobuf models, like `TW_Cosmos_Proto_StakeMessage`

File: swift/Tests/CosmosSignerTests.swift (hunk 1 above)
Comment (johnnynanjiang) [reply swift/Tests/CosmosSignerTests.swift]: done in 3a15f10707237aa97320142331a0a3171c3e8cdb

File: swift/Tests/CosmosSignerTests.swift (hunk 2 above)
Comment (johnnynanjiang) [reply swift/Tests/CosmosSignerTests.swift]: done in 3a15f10707237aa97320142331a0a3171c3e8cdb

**Pull request merged**